### ISSUE TITLE
Cluster update cannot be performed when using "proxy_server" cluster config option

### ISSUE THREAD

**Issue by lukeseawalker** [aws/aws-parallelcluster#3191]
Affected versions: >= 2.9.0 when using `scheduler=slurm` and `proxy_server` set in cluster configuration.

Cluster update cannot be performed because head node is not able to signal CloudFormation for update completion.

Workaround until a proper fix is released is to create the cluster with the following pre installation script

```
#!/bin/bash

. "/etc/parallelcluster/cfnconfig"

if [ "${cfn_node_type}" = "MasterServer" ]; then
cat << EOF > /etc/chef/cookbooks/aws-parallelcluster/templates/default/parallelcluster_supervisord.conf.erb
# Generated by Chef for AWS ParallelCluster <%= node['cfncluster']['cfn_node_type'] -%>
# Local modifications could be be overwritten.
<% case node['cfncluster']['cfn_node_type'] -%>
<%# MasterServer -%>
<% when 'MasterServer' -%>
<% if node['cfncluster']['cfn_scheduler'] == 'slurm' -%>
[program:cfn-hup]
command = bash -c "[ -f /etc/profile.d/proxy.sh ] && . /etc/profile.d/proxy.sh; <%= node['cfncluster']['cookbook_virtualenv_path'] %>/bin/python <%= node['cfncluster']['cookbook_virtualenv_path'] %>/bin/cfn-hup"
# The following are needed because cfn-hup starts as a daemon
exitcodes = 0
autorestart = unexpected
startsecs = 0

[program:clustermgtd]
command = <%= node['cfncluster']['node_virtualenv_path'] %>/bin/clustermgtd
redirect_stderr = true
stdout_logfile = /var/log/parallelcluster/clustermgtd

<% elsif node['cfncluster']['cfn_scheduler'] != 'awsbatch' -%>
[program:sqswatcher]
command = <%= node['cfncluster']['node_virtualenv_path'] %>/bin/sqswatcher
redirect_stderr = true
stdout_logfile = /var/log/sqswatcher

[program:jobwatcher]
command = <%= node['cfncluster']['node_virtualenv_path'] %>/bin/jobwatcher
redirect_stderr = true
stdout_logfile = /var/log/jobwatcher
<% end -%>

<% if node['cfncluster']['dcv_enabled'] == "master" && node['conditions']['dcv_supported'] -%>
[program:pcluster_dcv_authenticator]
command = <%= node['cfncluster']['dcv']['authenticator']['virtualenv_path'] %>/bin/python <%= node['cfncluster']['dcv']['authenticator']['user_home'] %>/pcluster_dcv_authenticator.py
      --port <%= Integer(node['cfncluster']['dcv_port']) + 1 %>
      --certificate <%= node['cfncluster']['dcv']['authenticator']['certificate'] %>
      --key <%= node['cfncluster']['dcv']['authenticator']['private_key'] %>
user = <%= node['cfncluster']['dcv']['authenticator']['user'] %>
environment = HOME="<%= node['cfncluster']['dcv']['authenticator']['user_home'] %>",USER="<%= node['cfncluster']['dcv']['authenticator']['user'] %>"
<% end -%>

<%# ComputeFleet -%>
<% when 'ComputeFleet' -%>
<% if node['cfncluster']['cfn_scheduler'] == 'slurm' -%>
[program:computemgtd]
command = <%= node['cfncluster']['node_virtualenv_path'] %>/bin/computemgtd
redirect_stderr = true
stdout_logfile = /var/log/parallelcluster/computemgtd
<% else -%>
[program:nodewatcher]
command = <%= node['cfncluster']['node_virtualenv_path'] %>/bin/nodewatcher
redirect_stderr = true
stdout_logfile = /var/log/nodewatcher
<% end -%>

<% end -%>
EOF
fi

```

**Comment by lukeseawalker** [comment]
this is available in [PCluster 3.0.0](https://github.com/aws/aws-parallelcluster/releases/tag/v3.0.0)

**Comment by enrico-usai** [comment]
Fixed for [2.11.4](https://github.com/aws/aws-parallelcluster/releases/tag/v2.11.4) too.
